src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,135 @@
+    mappings_df = mappings_df[mappings_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['object_prefix']
+
+    # Query synonyms
+    # - Mondo
+    mondo_df = pd.read_csv(mondo_scopes_and_xrefs_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    mondo_df['object_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['object_prefix']
+    mondo_df['unprefixed_source_id'] = mondo_df['source_id'].apply(lambda x: x.split(':')[1])
+    mondo_df = mondo_df[mondo_df['unprefixed_source_id'].str.isnumeric()]
+    del mondo_df['unprefixed_source_id']
+
+    # - Source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_df: pd.DataFrame = _get_synonyms(mappings_df['object_id'].tolist(), source_db)
+    source_df = source_df.rename(columns={'curie': 'source_id'})[['synonym_scope', 'synonym', 'source_id']]
+
+    # Determine synchronization cases
+    confirmed_df = mondo_df.merge(source_df, on=['synonym_scope', 'synonym', 'source_id'], how='inner')
+    confirmed_df['case'] = 'confirmed'
+
+    added_df = source_df.merge(mondo_df, on=['synonym_scope', 'synonym', 'source_id'], how='left', indicator=True)
+    added_df = added_df[added_df['_merge'] == 'left_only'].drop('_merge', axis=1)  # also can be done via Mondo ID = nan
+    added_df['case'] = 'added'
+
+    updated_df = mondo_df.merge(source_df, on=['synonym', 'source_id'], how='inner').rename(
+        columns={'synonym_scope_x': 'synonym_scope_mondo', 'synonym_scope_y': 'synonym_scope'})
+    updated_df = updated_df[updated_df['synonym_scope_mondo'] != updated_df['synonym_scope']]
+    updated_df['case'] = 'updated'
+
+    # TODO: -deleted
+    deleted_df = mondo_df.merge(source_df, on=['synonym_scope', 'synonym', 'source_id'], how='left', indicator=True)
+    deleted_df = deleted_df[deleted_df['_merge'] == 'left_only'].drop('_merge', axis=1)  # also can be done via Mondo ID = nan
+    deleted_df['case'] = 'deleted'
+    # TODO: filter these by where the entry does not exist in '-updated'
+    #  - because updated will appear as deletion
+    #  - how? i think: (mondo_id, synonym, source_id) appears in both
+    #  - how to drop? by index?
+    #  - is the below correct?
+    deleted_actually_updated_df = deleted_df.merge(updated_df, on=['mondo_id', 'synonym', 'source_id'], how='inner')
+
+    print()
+    # TODO: synonym type, e.g. DEPRACATED (sic)
+
+    # Write outputs
+    # todo: temp
+    dirty_df = pd.concat([confirmed_df, added_df, updated_df], ignore_index=True)[
+        ['mondo_id', 'synonym_scope', 'synonym_scope_mondo', 'synonym', 'source_id', 'case']]\
+        .sort_values(by=['case', 'synonym_scope', 'mondo_id', 'source_id', 'synonym'])
+    dirty_df.to_csv('~/Desktop/synonym_sync_combined_cases.tsv', sep='\t', index=False)
```

Comment:
## Current quick & dirty outputs
[Google sheet: `synonym_sync_combined_cases.tsv`](https://docs.google.com/spreadsheets/d/1Il3ikD9D4oKYcjuigFyFWMXJ7u7IeggDMrx6Rmdophk/edit?gid=1542962210#gid=1542962210)

**Cases**
Includes:
- `-confirmed`: Cases where the synonym still appears in the source on the given `source_id`, and the `synonym_scope` is the same.
- `-updated`: Cases where the synonym _scope_ is different between Mondo and the source (I haven't yet queried for _type_).
- `-added`: Cases where the the given synonym + `synonym_scope` does not exist in Mondo for the given `source_id`.
  - [ ] TODO: This needs to be filtered by some of the rows that appear in `-updated`. If the synonym exists in Mondo for a given `source_id`, but has a different `synonym_scope`, it should not show up in this case / file.

Not yet completed:
- `-deleted`: This was marked as a lower priority. I do currently have a rough draft of it done, but it suffers from the same issue as `-added. A lot of the rows in this file are simply cases where the `synonym_scope` is different, so I need to filter those out.

> [!note]
> `synonym_scope_mondo` column: This is only intended for the `-updated` template, so you can compare how the scope has changed. For the `-confirmed` template, the scope is the same in Mondo and the source. It's just not shown in those rows.

Reply:
@joeflack4 the Mondo ID column is empty in the Google Sheets.

Reply:
Bug. Will be fixed!

Reply:
Updated [Google sheet v2](https://docs.google.com/spreadsheets/d/1ftEZ47-JDe-1Z_fRWyO5NS39qT_-8MZsHXHEKz9aZ9Q/edit?gid=1786241482#gid=1786241482)!

That bug of the missing `mondo_id` in `-added` case is fixed.

Lots of quality checks went into the code for making this. I scanned it and it looks pretty good to me! Let me know what you think.